docs/plugins/backend-plugin.md

```diff
@@ -116,36 +97,172 @@ curl localhost:7007/api/carmen/health
 
 This should return `{"status":"ok"}` like before. Success!
 
+## Secure by Default
+
+In 1.25, Backstage started moving to a secure by default model for plugins. This means that network requests to plugins will by default not allow unauthenticated users. Let's take a deeper look at the above curl request which should allow unauthenticated access.
+
+The actual endpoint that is being called is defined in
+
+```ts title="plugins/carmen-backend/src/service/router.ts"
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { logger } = options;
+
+  const router = Router();
+  router.use(express.json());
+
+  // highlight-start
+  router.get('/health', (_, response) => {
+    logger.info('PONG!');
+    response.json({ status: 'ok' });
+  });
+  // highlight-end
+
+  router.use(errorHandler());
+  return router;
+}
+```
+
+You'll notice that there is no authentication mechanism defined here, just the route name and response data. That's because the authentication is handled in your plugin definition,
+
+```ts title="plugins/carmen-backend/src/plugin.ts"
+export const carmenPlugin = createBackendPlugin({
+  pluginId: 'carmenPlugin',
+  register(env) {
+    env.registerInit({
+      deps: {
+        httpRouter: coreServices.httpRouter,
+        logger: coreServices.logger,
+      },
+      async init({ httpRouter, logger }) {
+        httpRouter.use(
+          await createRouter({
+            logger,
+          }),
+        );
+        // highlight-start
+        httpRouter.addAuthPolicy({
+          path: '/health',
+          allow: 'unauthenticated',
+        });
+        // highlight-end
+      },
+    });
+  },
+});
+```
+
+This allows requests to this plugin's `/health` endpoint to go through unauthenticated!
+
+## Using Dependencies
+
+In the new backend, dependencies are defined statically during registration and then "injected" during initialization. Here's an example of what this looks like,
+
+```ts title="plugins/carmen-backend/src/plugin.ts"
+  register(env) {
+    env.registerInit({
+      // highlight-start
+      deps: {
+        httpRouter: coreServices.httpRouter,
+        logger: coreServices.logger,
+      },
+      // highlight-end
+      // And then you can use them through the options property!
+      // highlight-next-line
+      async init({ httpRouter, logger }) {
+        httpRouter.use(
+          await createRouter({
+            logger,
+          }),
+        );
+      },
+    });
+  },
+```
+
+You can add your own dependencies by adding a named item to the `deps` parameter:
+
+```ts
+deps: {
+  // highlight-next-line
+  myDependency: coreServices.rootConfig,
+},
+```
+
+And then you can access it by referencing it in the `init` block of your plugin definition,
+
+```ts
+async init({ myDependency }) {
+   // ..
+}
+```
+
+And then you're free to call it and pass it into your router as needed.
+
+Backstage provides a bunch of `coreServices` out of box:
```

Comment:
My main problem with the rest of the guide is that it duplicates a ton of information. I mean even before your changes. I'd suggest basically cutting out the entire rest of the article and linking to https://backstage.io/docs/backend-system/core-services/index and saying hey here's a huge toolbox, go there for more information about those facilities and use them as much as possible.

Then if THOSE docs are lacking, we can expand on them.

Does that make sense?

In particular when it comes to the user's identity at the bottom, it's something that we've identified as almost never even being wanted (there are just some absolute outliers that actually care at all), so having it mentioned in here is probably unfortunate.

Reply:
yup, is that something we can tackle in a more holistic revamp of these docs though? I'd like to get the initial migration through so folks can follow it, and we can start working on a new and improved version in the background.

Reply:
added the link out in [453f22d](https://github.com/backstage/backstage/pull/24069/commits/453f22dd4f452c86766b02e22af6fa22f107a4d3). The greater deduplication effort should be tracked as part of the docs updates.